lib/src/view_model.dart

```diff
@@ -15,16 +33,28 @@ abstract class ViewModel<State, Effect> {
 
   late final _effectController = StreamController<Effect>.broadcast();
 
+  /// The current [state]
   State get state => _state;
 
+  /// The last effect
   Effect? get lastEffect => _effect;
 
+  /// The state stream
+  ///
+  /// Will be canceled after [close] is called.
   Stream<State> get stateStream => _stateController.stream;
 
+  /// The effect stream
+  ///
+  /// Will be canceled after [close] is called.
   Stream<Effect> get effectStream => _effectController.stream;
 
+  /// Whether the [ViewModel] is closed.
+  ///
+  /// A [ViewModel] is considered closed once [close] is called.
   bool isClosed = false;
 
+  /// Emits a new [state]
```

Comment:
/// Emits a new [State] to the ViewModel.
///
/// This method is responsible for handling the emission of states. It checks
/// if the ViewModel is closed before attempting to emit a new state.
///
/// Parameters:
///   - [state]: The new state to be emitted by the ViewModel.
///
/// If the ViewModel is closed, a debug message is printed, and the state emission is skipped.
/// If the provided state is equal to the current state, the emission is also skipped.
/// Otherwise, the new state is set, and it is added to the state controller.
///
/// Throws:
///   - If an error occurs during the state emission, it is rethrown.